lms/djangoapps/courseware/views.py

```diff
@@ -1231,3 +1243,68 @@ def course_survey(request, course_id):
         redirect_url=redirect_url,
         is_required=course.course_survey_required,
     )
+
+
+def is_course_passed(course, grade_summary=None, student=None, request=None):
+    """
+    check user's course passing status. return True if passed
+
+    Arguments:
+        course : course object
+        grade_summary (dict) : contains student grade details.
+        student : user object
+        request (HttpRequest)
+
+    Returns:
+        returns bool value
+    """
+    nonzero_cutoffs = [cutoff for cutoff in course.grade_cutoffs.values() if cutoff > 0]
+    success_cutoff = min(nonzero_cutoffs) if nonzero_cutoffs else None
+
+    if grade_summary is None:
+        grade_summary = grades.grade(student, request, course)
+
+    return success_cutoff and grade_summary['percent'] > success_cutoff
+
+
+@ensure_csrf_cookie
+@require_POST
+def generate_user_cert(request, course_id):
+    """
+    It will check all validation and on clearance will add the new-certificate request into the xqueue.
+
+     Args:
+        request (django request object):  the HTTP request object that triggered this view function
+        course_id (unicode):  id associated with the course
+
+    Returns:
+        returns json response
+    """
+
+    if not request.user.is_authenticated():
+        log.info(u"Anon user trying to generate certificate for %s", course_id)
+        return JsonResponseBadRequest(_('You must be logged-in to generate certificate'))
+
+    student = request.user
+
+    course_key = CourseKey.from_string(course_id)
+
+    course = modulestore().get_course(course_key, depth=2)
+    if not course:
+        return JsonResponseBadRequest(_("Course is not valid"))
+
+    if not is_course_passed(course, None, student, request):
+        return JsonResponseBadRequest(_("You failed to pass the course."))
+
+    certificate_status = certificate_downloadable_status(student, course.id)
+
+    if not certificate_status["is_downloadable"] and not certificate_status["is_generating"]:
+        generate_user_certificates(student, course)
+        return JsonResponse(_("Certificate is generating."))
+
+    # if certificate_status is not is_downloadable and is_generating or
+    # if any error appears during certificate generation return the message cert is generating.
+    # with badrequest
+    # at backend debug the issue and re-submit the task.
+
+    return JsonResponseBadRequest(_("Certificate is generating."))
```

Comment:
Are JSON-encoded responses necessary in this view? The response content is a string.